assets/js/base/utils/i18n.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable no-console */
+/**
+ * External dependencies
+ */
+import { setLocaleData } from '@wordpress/i18n';
+import { fetchDedupe } from 'fetch-dedupe';
+
+interface RequireChunkCallback {
+	add: (
+		callback: (
+			props: { scriptSrc: string; publicPath: string },
+			promises: Promise< unknown >[]
+		) => void
+	) => this;
+}
+
+declare global {
+	interface Window {
+		__requireChunkCallback__: RequireChunkCallback;
+	}
+}
+
+/**
+ * Adds require chunk handler for fetching translations.
+ */
+export const addRequireChunkTranslationsHandler = (
+	options: {
+		// Callback function to get the file name from the ChunkId.
+		getTranslationChunkFileUrl?: ( chunkId: string ) => string;
+		// Text domain.
+		domain?: string;
+		// Array of chunk ids that should have available translations.
+		translatedChunks?: string[];
+	} = {}
+): void => {
+	if ( typeof window.__requireChunkCallback__ === undefined ) {
```

Comment:
The result of `typeof` is a string, like `"undefined"`. Comparing to the `undefined` value will always yield `false`.